### PR TITLE
Handle interpreter change

### PR DESCRIPTION
**Pull request by MikhailArkhipov**
- Send updates to Pylance
- Minor reformatting to more modern style

### CODE REVIEW

File: Python/Product/PythonTools/PythonTools/LanguageServerClient/PythonLanguageClient.cs [hunk 1]
```diff
@@ -103,33 +94,34 @@ public class PythonLanguageClient : ILanguageClient, ILanguageClientCustomMessag
...
-            }
+            _clientContext = PythonWorkspaceContextProvider.Workspace != null 
+                ? (IPythonLanguageClientContext)new PythonLanguageClientContextWorkspace(PythonWorkspaceContextProvider.Workspace, PythonCoreConstants.ContentType) 
+                : new PythonLanguageClientContextGlobal(OptionsService, PythonCoreConstants.ContentType);
+
+            _clientContext.InterpreterChanged += OnInterpreterChanged;
+            _clientContext.SearchPathsChanged += OnSearchPathsChanged;
+            _disposables.Add(() => {
+                _clientContext.InterpreterChanged -= OnInterpreterChanged;
+                _clientContext.SearchPathsChanged -= OnSearchPathsChanged;
```
Comment (huguesv) [inline Python/Product/PythonTools/PythonTools/LanguageServerClient/PythonLanguageClient.cs:105]: I've noticed that we don't have anyone disposing `_clientContext`, I think it should be added here after the event unregistration.

File: Python/Product/PythonTools/PythonTools/LanguageServerClient/PythonLanguageClientContextGlobal.cs [hunk 2]
```diff
@@ -24,20 +24,15 @@ namespace Microsoft.PythonTools.LanguageServerClient {
...
 #pragma warning disable CS0067
         public event EventHandler SearchPathsChanged;
         public event EventHandler Closed;
 #pragma warning restore CS0067
 
-        public PythonLanguageClientContextGlobal(
-            IInterpreterOptionsService optionsService,
-            string contentTypeName
-        ) {
+        public PythonLanguageClientContextGlobal(IInterpreterOptionsService optionsService, string contentTypeName) {
```
Comment (huguesv) [inline Python/Product/PythonTools/PythonTools/LanguageServerClient/PythonLanguageClientContextGlobal.cs:35]: Btw, the contentTypeName here and in other places is a leftover from MPLS where each language server instance had its own derived content type in order to associate a file with the correct server.

We no longer need this, so if you feel like it, you can remove it from here, and the IPythonLanguageClientContext interface.

Comment (MikhailArkhipov) [reply Python/Product/PythonTools/PythonTools/LanguageServerClient/PythonLanguageClientContextGlobal.cs]: Yeah, I removed it. I was also wondering why it was there... :-)

File: Python/Product/PythonTools/PythonTools/LanguageServerClient/PythonLanguageClient.cs (hunk 1 above)
Comment (MikhailArkhipov) [reply Python/Product/PythonTools/PythonTools/LanguageServerClient/PythonLanguageClient.cs:105]: Yep, added disposing.

**Review by huguesv — APPROVED** [review]

**Pull request merged**